### DISCUSSION TITLE
When creating a new repo, `push an existing repository from the command line` command that GitHub suggests to copy and paste into terminal should fail if `origin` remote already exists, so it doesn't unexpectedly push to a repo that the user was not expecting

### DISCUSSION THREAD

**Discussion by msabramo** [community/community#77577] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

I just had an unfortunate thing happen. I cloned a second copy of a github repo to start a new project and in the new one I did major changes to the directory structure. Then I was ready to push the new repo. In the github UI, I created the new repo and then it suggested that I could `push an existing repository from the command line` by copying the following command and pasting it into my terminal:

```
git remote add origin git@github.com:msabramo/test001.git
  git branch -M main
  git push -u origin main
```

Now I've copied and pasted commands like that hundreds of times. I don't read it carefully or stop to think about it, because it's worked perfectly for me hundreds of times... Until now... because this is a special case...

I executed the above command in my new repo and the results were disastrous -- not disastrous for the new repo but disastrous for the old repo, which the whole team relies on... (easy to fix but caused a few minutes of panic). It looked something like this:

```
$ git remote add origin git@github.com:msabramo/test001.git
git branch -M main
git push -u origin main
error: remote origin already exists.
Enumerating objects: 51, done.
Counting objects: 100% (51/51), done.
Delta compression using up to 10 threads
Compressing objects: 100% (43/43), done.
Writing objects: 100% (48/48), 1.59 MiB | 4.38 MiB/s, done.
Total 48 (delta 13), reused 6 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (13/13), completed with 2 local objects.
To github.com:msabramo/my-original-project.git
   b837d022..b85ad2b6  main -> main
branch 'main' set up to track 'origin/main'.
```

You see because I started this project by cloning from the old one, this project already has an `origin` remote pointing to the old repo. So when I invoked the above commands in my shell, the `git remote add origin ...` command failed because the remote already exists, but then it kept going executing the other 2 commands and it resulted in pushing commits to my old repository.

Again I was able to fix this by resetting `main` in the old repository, but it was stressful, so I am proposing that the suggested command be changed to be more robust:

```
git remote add origin git@github.com:msabramo/test001.git && \
  git branch -M main && \
  git push -u origin main
```

With this new version, if one command fails then the others will not execute. This would've prevented the problem I experienced, because the `git remote add origin ...` command would've failed and then it wouldn't do the other 2 commands. It would just fail hard and I'd figure out the problem and fix it and not mess up another repo unexpectedly. Here's a demo:

```
$ git remote add origin git@github.com:msabramo/test001.git && \
  git branch -M main && \
  git push -u origin main
error: remote origin already exists.
```

What do you think?

**Comment by shyakachaste** [comment]
Your proposed modification to the command is a sensible precautionary measure that can help prevent unintended consequences, especially in scenarios where a repository might already have an existing remote. By combining the commands with &&, you ensure that the subsequent commands are only executed if the preceding ones succeed, reducing the risk of pushing changes to the wrong repository. The suggested modification adds an extra layer of safety by stopping the execution chain upon encountering an error, making it a more robust practice. Implementing this change in your workflow can indeed help avoid potential issues, and it's a good practice to consider in similar situations where multiple commands are executed sequentially.

**Comment by msabramo** [comment]
Yep. I think GitHub should change the default command they present to what I showed. That way people can copy and paste and not have to worry about nasty stuff happening.

**Comment by shyakachaste** [comment, marked as answer]
GitHub should adopt a safer default command by including checks with && to prevent unintended consequences during repository setup, offering users a more secure and user-friendly experience.

**Comment by AhsanAlijs** [comment]
Your proposed change to the suggested command is a reasonable improvement to avoid potential issues, especially in cases where the user might not be aware of the existing remote. Adding a conditional execution based on the success of the previous command can prevent unintended consequences.

Your modified command:

bash

git remote add origin git@github.com:msabramo/test001.git && \
git branch -M main && \
git push -u origin main
This version ensures that if the first command fails (due to the remote already existing), the subsequent commands won't be executed. It's a safer approach to prevent accidental pushes to the wrong repository.

Consider submitting this feedback to GitHub or the relevant platform where you encountered the issue. Many developers appreciate constructive feedback, and it might contribute to the improvement of user experience for others as well.
